go/buildhelper/lib.go

```diff
@@ -0,0 +1,43 @@
+package buildhelper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+var CONNECTION_TIMEOUT = 15 * time.Second
+
+// EthAPI struct
+// based on https://eth.wiki/json-rpc/API
+type EthAPI struct {
+	ApiClient *ethclient.Client
+	retry     int
+	timeout   time.Duration
+	ipaddress string
```

Comment:
@otherview Why put the `ipaddress` and `port` on the struct? Wouldn't it make more sense to have them as `Connect` params (so the struct isn't coupled to what you connected to)?

Reply:
First pass, have to clean up. But ideally, it would be 1 instance of the object per connection.
If there is a need to have a different connection you instantiate a new object.

Reply:
Outdated :)